File: 168.Excel-Sheet-Column-Title.go
// https://leetcode.com/problems/excel-sheet-column-title/
//
// algorithms
// Easy (28.32%)
// Total Accepted:    159,684
// Total Submissions: 563,827
// beats 100.0% of golang submissions

package leetcode

func convertToTitle(n int) string {
	var res []byte

	for n > 0 {
		n -= 1
		remainder := n % 26
		res = append([]byte{byte('A' + remainder)}, res...)
		n /= 26
	}

	return string(res)
}

File: 168.Excel-Sheet-Column-Title.java
// https://leetcode.com/problems/excel-sheet-column-title/
//
// algorithms
// Easy (30.19%)
// Total Accepted: 200,464
// Total Submissions: 663,997
// beats 100.0% of java submissions


class Solution {
    public String convertToTitle(int n) {
        StringBuilder sb = new StringBuilder();

        while (n > 0) {
            int tmp = (n - 1) % 26;
            sb.append((char) ('A' + tmp));

            n = (n - 1) / 26;
        }

        return sb.reverse().toString();
    }
}